server/controllers/following.js

```diff
@@ -0,0 +1,114 @@
+const Following = require("../models/Following");
+const asyncHandler = require("express-async-handler");
+const User = require("../models/User");
+const smartSearch = require("../utils/smartSearch");
+
+// @route GET /following
+// @desc get user's following list
+// @access Private
+exports.getFollowing = asyncHandler(async (req, res, next) => {
+	const data = await Following.findOne(
+		{ userId: req.user.id },
+		{ _id: 0, following: 1 }
+	).populate("following", "_id username image");
+
+	//in case the user has not followed anyone yet
+	if (!data) {
+		res.status(200).json({
+			success: {
+				followingList: [],
+			},
+		});
+	}
+	res.status(200).json({
+		success: {
+			followingList: data.following,
+		},
+	});
+});
+
+// @route GET /following/peopleYouMightKnow
+// @desc returns a list of the peaple the user might know
+// @access Private
+exports.getPeopleYouMightKnow = asyncHandler(async (req, res, next) => {
+	const userRelations = await Following.findOne({ userId: req.user.id }).populate(
+		"followers",
+		"_id username image"
+	);
+
+	// returns the users that follow the user but the user does not follow back and
+	// a list of users that should be excluded from our broader search
+	const { excludeFromSearch, noFollowBack } = smartSearch(userRelations, req.user.id);
+
+	//returns the users that their id does not exist in the excludedFromSearch list
+	const searchResult = await User.find(
+		{
+			_id: {
+				$nin: excludeFromSearch,
+			},
+		},
+		"_id username image"
+	);
+
+	//In the final result the people who follow the user but the user
+	//does not follow back will be at the top of the list, which leads to
+	//an smarter result for the people you might know section of the app
+	const finalResponse = [...noFollowBack, ...searchResult];
+
+	res.json(finalResponse);
+});
+
+// @route POST /following/followOrUnfollow
```

Comment:
just POST `follow` instead of `followOrUnfollow` is more typical route name

Reply:
changed to follow